modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java

```diff
@@ -1093,6 +1101,52 @@ private boolean _isActiveProfile(Project project, String profileName) {
 		return false;
 	}
 
+	private void _mapGlobalJSScriptElementAttributesToString(
+		ClientExtension clientExtension) {
+
+		Map<String, Object> typeSettings = clientExtension.typeSettings;
+
+		Map<String, Object> scriptElementAttributes =
+			(LinkedHashMap<String, Object>)typeSettings.get(
+				"scriptElementAttributes");
+
+		if ((scriptElementAttributes == null) ||
+			scriptElementAttributes.isEmpty()) {
+
+			typeSettings.remove("scriptElementAttributes");
+
+			return;
+		}
+
+		StringBuilder stringBuilder = new StringBuilder();
+
+		int i = 0;
+
+		for (Map.Entry<String, Object> entry :
+				scriptElementAttributes.entrySet()) {
+
+			stringBuilder.append(entry.getKey());
+			stringBuilder.append("=\"");
+
+			Object value = entry.getValue();
+
+			if (value == null) {
+				value = true;
+			}
+
+			stringBuilder.append(value);
+			stringBuilder.append("\"");
+
+			if (i != (scriptElementAttributes.size() - 1)) {
+				stringBuilder.append(StringPool.NEW_LINE);
+			}
```

Comment:
Let's avoid this index check and instead use either an array/collection and `StringUtil.join`, or a stream.

Reply:
`StringUtil.join` is not enough. I need to check if the original value is `null` and append `attribute="value"`

I also can't see how  `Stream` would help in this case, I guess this would fall at the same condition (is the last index?).

I guess I'm overlooking something here...

Reply:
> StringUtil.join is not enough. I need to check if the original value is null and append attribute="value"

Why would the value be null? Wouldn't `yaml` complain beforehand? And wouldn't DXP handle setting whatever default value is needed for an absent property?

Reply:
> I also can't see how Stream would help in this case, I guess this would fall at the same condition (is the last index?).

You could use stream to map each entry to whatever string value you wanted, then join them with `Collectors.joining("\n")`.

Reply:
> You could use stream to map each entry to whatever string value you wanted, then join them with Collectors.joining("\n").

Of course 🤦